src/typedefs.ts

```diff
@@ -131,7 +131,7 @@ export type TEvent<E extends Event = TPointerEvent> = {
   e: E;
 };
 
-export type TransformEvent = TEvent & {
+export type TransformEvent<E extends Event = TPointerEvent> = TEvent<E> & {
```

Comment:
I think all the types related to events should be extracted from this file
Didn't want to cause too many merge conflicts in parallel PRs